src/dparse/lexer.d

```diff
@@ -388,8 +389,8 @@ public bool isLiteral(IdType type) pure nothrow @safe @nogc
  * whitespace tokens are skipped and comments are attached to the token nearest
  * to them.
  */
-const(Token)[] getTokensForParser(ubyte[] sourceCode, LexerConfig config,
-    StringCache* cache)
+const(Token)[] getTokensForParser(R)(R sourceCode, LexerConfig config, StringCache* cache)
+if (is(Unqual!(ElementEncodingType!R) : ubyte) && isDynamicArray!R)
```

Comment:
Why not relax `isDynamicArray` to `isRandomAccessRange`?

Reply:
At the end LexerRange stores a `const(ubyte)[]` so it cannot be something else than a dynamic array.

Reply:
Actually this could always be tweaked but it's really not worth.

Reply:
Fair enough.